helpers/authentication.js

```diff
@@ -131,33 +131,23 @@ const populateCurrentUser = async (req, res) => {
 			return Promise.resolve(res.locals.currentSchoolData);
 		}
 		return Promise.all([
-			api(req).get(`/users/${payload.userId}`),
-			api(req).get(`/roles/user/${payload.userId}`),
-		]).then(([user, roles]) => {
-			const data = {
-				...user,
-				roles,
-				permissions: roles.reduce((acc, role) => [...new Set(acc.concat(role.permissions))], []),
-			};
-			res.locals.currentUser = data;
-			setTestGroup(res.locals.currentUser);
-			res.locals.currentRole = rolesDisplayName[data.roles[0].name];
-			res.locals.roles = data.roles.map(({ name }) => name);
-			res.locals.roleNames = data.roles.map((r) => rolesDisplayName[r.name]);
-			return api(req, { version: 'v3' }).get(`/school/id/${res.locals.currentUser.schoolId}`).then((data2) => {
-				res.locals.currentSchool = res.locals.currentUser.schoolId;
-				res.locals.currentSchoolData = data2;
-				res.locals.currentSchoolData.isExpertSchool = data2.purpose === 'expert';
-				return data2;
-			});
-		}).catch((e) => {
-			// 400 for missing information in jwt, 401 for invalid jwt, not-found for deleted user
-			if (e.statusCode === 400 || e.statusCode === 401 || e.error.className === 'not-found') {
```

Comment:
Den catch hier raus zu nehmen halte ich für gefährlich. 
Ich verstehe nicht ganz warum er existiert, aber er entfernt immerhin den cockie wenn ein NotFound | 400 | 401 von get users , get roles, get school kommt. Das klinkt erstmal nicht so verkehrt. Ich würde vermuten das man eher für jeden Fehler auch JS Fehler dann den cockie entfernen möchte, aber ihn stehen zu lassen halt ich auch für falsch. 

Habe mir allerdings auc nicht den umliegenden Code angeschaut.

Reply:
Gehst du gerade die einzelnen Commits durch? Weil diese Stelle wurde temporär als Change angezeigt, weil da jemand die Formatierung geändert hat. Ich hab das aber wieder angeglichen. Guck lieber in die geänderten Files!

Reply:
Ich schaue mir immer die Änderungen seit dem letzten Review an.

Reply:
Ah okay. Dann guckst du dir aber auch alles mit an, was rein gemerget wurde. Das war jetzt hier der Fall.